### PR TITLE
Add CI job to push Docker image to Docker Hub

### LINKED ISSUE #31: Publish Docker image on Docker Hub
Add a workflow `master-push.yml` with a job that builds the image and pushes it to Docker Hub.

The image is tested in: https://github.com/runtimeverification/kontrol/blob/031f4b046726edf8eb05b84cda3379f6d7f1db50/.github/workflows/test-pr.yml#L125

### PR DESCRIPTION
**Pull request by tothtamas28**
Fixes #31

### CODE REVIEW

**Comment by tothtamas28** [comment]
@ehildenb or @F-WRunTime, please add `DOCKERHUB_PASSWORD` to `secrets`.

https://hub.docker.com/r/runtimeverificationinc/kontrol might also need to be set up for `docker image push` to go through.

**Comment by F-WRunTime** [comment]
> @F-WRunTime, please add `DOCKERHUB_PASSWORD` to `secrets`.
> 
> https://hub.docker.com/r/runtimeverificationinc/kontrol might also need to be set up for `docker image push` to go through.

done

**Comment by ehildenb** [comment]
@tothtamas28 we also need to be:

- Publishing the Nix binaries to the public nix cache on master pushes.
- Cutting a GitHub release with the version tag (so that we can do `kup install kontrol --version VERSION_TAG`).

**Review by ehildenb — APPROVED** [review]

**Comment by F-WRunTime** [comment]
> @tothtamas28 we also need to be:
> 
> * Publishing the Nix binaries to the public nix cache on master pushes.
> * Cutting a GitHub release with the version tag (so that we can do `kup install kontrol --version VERSION_TAG`).

Addressing the credentials needed for this. CACHIX_PUBLIC_TOKEN is available for pushing to the public cachix binary store.

**Comment by tothtamas28** [comment]
> we also need to be

Opened issues:
* https://github.com/runtimeverification/kontrol/issues/36
* https://github.com/runtimeverification/kontrol/issues/37

File: .github/workflows/master-push.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: 'Master Push'
+on:
+  push:
+    branches:
+      - 'master'
+concurrency:
+  group: ${{ github.workflow }}
+  cancel-in-progress: true
```
[the remark this answers is not in the document]
Comment (tothtamas28) [reply .github/workflows/master-push.yml]: Good point, done.

**Pull request merged**